### PR TITLE
chore(unlock-js,unlock-app,locksmith): upgrade to ethers 6

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This replaces ethers v5 by v6

## Unlock JS

Before the changes 

```

 Test Files  6 passed (6)
      Tests  11219 passed | 168 skipped (11387)
   Start at  15:23:11
   Duration  1581.31s (transform 396ms, setup 0ms, collect 1.94s, tests 1580.36s, environment 1ms, prepare 315ms)
```

After the changes 

```

 Test Files  6 passed (6)
      Tests  11219 passed | 168 skipped (11387)
   Start at  14:57:09
   Duration  1399.96s (transform 434ms, setup 0ms, collect 2.47s, tests 1398.92s, environment 1ms, prepare 241ms)
```

## Unlock App

Before 
```
 Test Files  28 passed | 2 skipped (30)
      Tests  228 passed | 13 skipped (241)
   Start at  06:03:34
   Duration  21.31s (transform 1.36s, setup 867ms, collect 12.76s, tests 22.12s, environment 15.45s, prepare 3.90s)
```

After 

```

```

### CODE REVIEW

File: packages/unlock-js/examples/purchase-key.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ const ethers = require('ethers')
 const { WalletService } = require('../lib/index')
 const networks = require('./networks')
 
-const provider = new ethers.providers.JsonRpcBatchProvider(networks[4].provider)
```
Comment (julien51) [inline packages/unlock-js/examples/purchase-key.js:6]: The batch provider was pretty cool as it made the app a bit faster by "bundling" multiple RPC calls into a single one. Is that not a thing anymore?

Comment (clemsos) [reply packages/unlock-js/examples/purchase-key.js:6]: the batch mode is actually enabled by default in ethers v6 https://docs.ethers.org/v6/api/providers/jsonrpc/#JsonRpcApiProviderOptions

default options are 

https://github.com/ethers-io/ethers.js/blob/92761872198cf6c9334570da3d110bca2bafa641/src.ts/providers/provider-jsonrpc.ts#L205-L207

Comment (julien51) [reply packages/unlock-js/examples/purchase-key.js:6]: Very cool!

**Comment by julien51** [comment]
Hum. It looks like we will have to do `locksmith` as well :/

**Pull request merged**